Add a tool to take an inverted screenshot

Cr4ckC4t opened the pull request:
## Why invert colors?

When using flameshot in dark environments on a regular basis it would be very useful to invert the colors in the capture, making the screenshot white(ish) background and dark text. Thus, enhancing the usability of those screenshots in reports and prints without having to use additional tools or workarounds.

Also, it's been a feature request for quite a while (#689).

## How does it work?

Basically, I copied the `savetool` and made an `inverttool`, that inverses all pixels before saving the screenshot. The actual consists of four lines:
```
    QPixmap inverted = context.selectedScreenshotArea();
    QImage img = inverted.toImage();
    img.invertPixels();
    inverted.convertFromImage(img);
```

However, it turned out that flameshot requires to update quite a few places in order to add a tool. Without much of a documentation I tried to stay close to how the `savetool` is setup.

## Before merging

I'd advise that someone with a way better understanding of this project looks at my modifications in order to spot potentially missing or even unnecessary code for this new tool to be integrated.

## Example

See the above mentioned issue for a video demo. Here's a screenshot of the newly added button.
![invert-example](https://user-images.githubusercontent.com/63863112/131217162-43633f2d-abe9-462c-887b-dc30e13f4476.PNG)

## Usage

After choosing the area and drawing markers, text, arrows etc. - use `Ctrl+i` or the *Invert* button to save the screenshot with inverted colors.

## Notes

This isn't a perfect solution. People might want to invert colors while still drawing stuff or see a review of the inversion before saving the capture. Additionally, more filters could be added maybe ([comment by @mmahmoudian](https://github.com/flameshot-org/flameshot/issues/689#issuecomment-907575403)). However, my Qt knowledge isn't by far good enough to be able to help with those things.

From my point of view, this Invert tool is a good solution for the time being and could still be extended at a later point.

---
Alright, that's it - let me know if I missed something or should update anything.

borgmanJeremy commented:
What do you think about changing it so a rectangular region is inverted instead of the entire section? I think that will allow it to be more flexible

Cr4ckC4t commented:
That's an interesting idea. For my workflow, saving the inverted screenshot is good enough but I can see how that would make the tool more flexible.

Currently I have no idea how to do that but I will try and see if I can implement your suggestion and report back here.

borgmanJeremy commented:
I'm happy to provide guidance if you have questions or want help! Maybe start by looking at the pixelate tool, this will be similar.

Cr4ckC4t commented:
Haha, that was exactly my thought. (Thanks for the offer.)

So I came up with this:

https://user-images.githubusercontent.com/63863112/131228085-8ff89150-3278-40a9-b2ad-7bebe469bc3e.mp4

Does that match your idea?

It's basically just a stripped down version of pixelate with inversion instead of bluring.

borgmanJeremy commented:
Perfect!

Cr4ckC4t commented:
So I'm unsure about one thing. The pixelate tool used a scaled selection by querying the pixel-aspect ratio of the screen before copying the screen area. However, the normal rectangle tool and selection tool don't do that. I've left that out for now and used the basic selection area (`point().first` to `point().second`) which seems to work just fine. I don't know if the scaled selection maybe is necessary for different screen resolutions or if that was just for pixelating- so maybe check that (`inverttool.cpp`).

Other than that, I've updated the butten order of the invert tool (next to pixelate). While doing that I also noticed that the numbering tool is probably supposed to be with the other drawing tools as well, so I positioned it next to the inverter too (let me know if that's not wanted).

To match the other drawing tools I also updated the description and the shortcut (now just `i`).

Here's what it looks like now:
![invert-example](https://user-images.githubusercontent.com/63863112/131245786-0578084d-c15e-4309-8a5e-c14e183575a8.PNG)

borgmanJeremy commented on src/tools/invert/inverttool.cpp:
```diff
@@ -0,0 +1,84 @@
+void InvertTool::drawSearchArea(QPainter& painter, const QPixmap& pixmap)
+{
+    Q_UNUSED(pixmap)
+    Q_UNUSED(painter)
```
Reference the implementation in the Pixelate tool for this. Currently since you marked Q_UNUSED(painter), this new feature does not tie into the "movable" feature. IE I cannot move the selection once placed. 

If you duplicate the implementation that is in the pixelate feature that will resolve this.

Cr4ckC4t replied:
✅ Done in `9889eadf`, although I didn't fully understand. I was able to move the screen selection without those lines too. 

Regardless, it's now the same as in the `pixelate tool`.

borgmanJeremy replied:
Thanks, what I meant was I could not move the inverted box the way other objects are moveable. This should fix it.

Cr4ckC4t commented on data/translations/Internationalization_ca.ts:
```diff
@@ -1303,6 +1303,19 @@ You may need to escape the &apos;#&apos; sign as in &apos;\#FFF&apos;</source>
         <translation>Guarda la captura</translation>
     </message>
 </context>
+<context>
```
✅ Changes reverted in `973aef74` and `de41c958`.

Cr4ckC4t commented on data/img/material/black/invert.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1025" height="1024" preserveAspectRatio="xMidYMid meet" style="transform: rotate(360deg);"><rect id="backgroundrect" width="100%" height="100%" x="0" y="0" fill="none" stroke="none"/><g class="currentLayer" style=""><title>Layer 1</title><path d="M896.428 1024h ...
```
I did not. I don't actually remember the site I got it from.

Instead, I've seen that flameshot already uses Material Design icons so I updated the icon to [this one](https://materialdesignicons.com/icon/invert-colors) (black and white). The current license file covers this already if I'm not mistaken.

✅ `c420842b`

Creteil commented:
Hi all,

Sorry for my ignorance, but from where I can fetch this source code to have access to this latest features commits ?

Creteil commented:
Look like it must appear after « change request » approval in the master branch...

True ?

If so, sorry to bogus on the question...

Cr4ckC4t commented:
@Creteil you could clone my fork from here: https://github.com/Cr4ckC4t/flameshot/tree/invert (branch: `invert`). Building this flameshot allows you to use the `invert` tool.

However, maybe this gets merched soon (thanks to @borgmanJeremy for the quick responses and reviews) - so I'd say keep an eye on the original flameshot:master.

borgmanJeremy reviewed (approved):
Looks great! Thanks for the contribution!
